components/needs_team/needs_team.jsx

```diff
@@ -80,6 +82,21 @@ export default class NeedsTeam extends React.Component {
 
         const team = this.updateCurrentTeam(this.props);
 
+        DesktopBridge.connect((event, data) => {
+            switch (event) {
+            case 'updateUserActivityStatus':
+                // ping the server if the desktop app reports the user is still active
+                if (this.props.currentUser && data.userIsActive === true) {
+                    this.userIsActive = true;
```

Comment:
It's a bit unfortunate to have another `userIsActive` bit in the system. 0/5 on whether it needs to happen now, but I wonder if we could expose a new action that itself embodies the logic of checking the old state and deciding whether to call `viewChannel`, abstracting the "hack" into a higher level. Then presumably the existing code that monitors focus/blur on the window could hook into the same action.

Reply:
That is not really needed for the initial version of this functionality – has been removed. There will possibly be a next version with some enhancements that could look into something like this if needed.